File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-main-5-4/ValuationDocument.cpp
// ValuationDocument.cpp 
#include "ValuationDocument.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ValuationDocument::ValuationDocument(TiXmlNode* xmlNode)
: DataDocument(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //marketNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* marketNode = xmlNode->FirstChildElement("market");

   if(marketNode){marketIsNull_ = false;}
   else{marketIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- marketNode , address : " << marketNode << std::endl;
   #endif
   if(marketNode)
   {
      if(marketNode->Attribute("href") || marketNode->Attribute("id"))
      {
          if(marketNode->Attribute("id"))
          {
             marketIDRef_ = marketNode->Attribute("id");
             market_ = boost::shared_ptr<Market>(new Market(marketNode));
             market_->setID(marketIDRef_);
             IDManager::instance().setID(marketIDRef_,market_);
          }
          else if(marketNode->Attribute("href")) { marketIDRef_ = marketNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { market_ = boost::shared_ptr<Market>(new Market(marketNode));}
   }

   //valuationSetNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* valuationSetNode = xmlNode->FirstChildElement("valuationSet");

   if(valuationSetNode){valuationSetIsNull_ = false;}
   else{valuationSetIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- valuationSetNode , address : " << valuationSetNode << std::endl;
   #endif
   if(valuationSetNode)
   {
      if(valuationSetNode->Attribute("href") || valuationSetNode->Attribute("id"))
      {
          if(valuationSetNode->Attribute("id"))
          {
             valuationSetIDRef_ = valuationSetNode->Attribute("id");
             valuationSet_ = boost::shared_ptr<ValuationSet>(new ValuationSet(valuationSetNode));
             valuationSet_->setID(valuationSetIDRef_);
             IDManager::instance().setID(valuationSetIDRef_,valuationSet_);
          }
          else if(valuationSetNode->Attribute("href")) { valuationSetIDRef_ = valuationSetNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { valuationSet_ = boost::shared_ptr<ValuationSet>(new ValuationSet(valuationSetNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Market> ValuationDocument::getMarket()
{
   if(!this->marketIsNull_){
        if(marketIDRef_ != ""){
             return boost::shared_static_cast<Market>(IDManager::instance().getID(marketIDRef_));
        }else{
             return this->market_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Market>();
   }
}
boost::shared_ptr<ValuationSet> ValuationDocument::getValuationSet()
{
   if(!this->valuationSetIsNull_){
        if(valuationSetIDRef_ != ""){
             return boost::shared_static_cast<ValuationSet>(IDManager::instance().getID(valuationSetIDRef_));
        }else{
             return this->valuationSet_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ValuationSet>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-main-5-4/ValuationDocument.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ValuationDocument : DataDocument
    {
        public ValuationDocument(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode marketNode = xmlNode.SelectSingleNode("market");
            
            if (marketNode != null)
            {
                if (marketNode.Attributes["href"] != null || marketNode.Attributes["id"] != null) 
                {
                    if (marketNode.Attributes["id"] != null) 
                    {
                        marketIDRef_ = marketNode.Attributes["id"].Value;
                        Market ob = new Market(marketNode);
                        IDManager.SetID(marketIDRef_, ob);
                    }
                    else if (marketNode.Attributes["href"] != null)
                    {
                        marketIDRef_ = marketNode.Attributes["href"].Value;
                    }
                    else
                    {
                        market_ = new Market(marketNode);
                    }
                }
                else
                {
                    market_ = new Market(marketNode);
                }
            }
            
        
            XmlNode valuationSetNode = xmlNode.SelectSingleNode("valuationSet");
            
            if (valuationSetNode != null)
            {
                if (valuationSetNode.Attributes["href"] != null || valuationSetNode.Attributes["id"] != null) 
                {
                    if (valuationSetNode.Attributes["id"] != null) 
                    {
                        valuationSetIDRef_ = valuationSetNode.Attributes["id"].Value;
                        ValuationSet ob = new ValuationSet(valuationSetNode);
                        IDManager.SetID(valuationSetIDRef_, ob);
                    }
                    else if (valuationSetNode.Attributes["href"] != null)
                    {
                        valuationSetIDRef_ = valuationSetNode.Attributes["href"].Value;
                    }
                    else
                    {
                        valuationSet_ = new ValuationSet(valuationSetNode);
                    }
                }
                else
                {
                    valuationSet_ = new ValuationSet(valuationSetNode);
                }
            }
            
        
        }
        
    
        #region Market_
        private Market market_;
        public Market Market_
        {
            get
            {
                if (this.market_ != null)
                {
                    return this.market_; 
                }
                else if (this.marketIDRef_ != null)
                {
                    market_ = IDManager.getID(marketIDRef_) as Market;
                    return this.market_; 
                }
                else
                {
                      return this.market_; 
                }
            }
            set
            {
                if (this.market_ != value)
                {
                    this.market_ = value;
                }
            }
        }
        #endregion
        
        public string marketIDRef_ { get; set; }
        #region ValuationSet_
        private ValuationSet valuationSet_;
        public ValuationSet ValuationSet_
        {
            get
            {
                if (this.valuationSet_ != null)
                {
                    return this.valuationSet_; 
                }
                else if (this.valuationSetIDRef_ != null)
                {
                    valuationSet_ = IDManager.getID(valuationSetIDRef_) as ValuationSet;
                    return this.valuationSet_; 
                }
                else
                {
                      return this.valuationSet_; 
                }
            }
            set
            {
                if (this.valuationSet_ != value)
                {
                    this.valuationSet_ = value;
                }
            }
        }
        #endregion
        
        public string valuationSetIDRef_ { get; set; }
        
    
        
    
    }
    
}

